BlueprintUI/Sources/Element/ComposedElement.swift

```diff
@@ -0,0 +1,22 @@
+import CoreGraphics
+
+/// An element built out of other elements.
+///
+/// Similar in purpose to `ProxyElement`, but the contents may change depending on the `Environment`.
+///
+/// - seealso: ProxyElement
+public protocol ComposedElement: Element {
+
+    /// Return the contents of this element in the given environment.
+    func elementRepresentation(in environment: Environment) -> Element
+}
+
+extension ComposedElement {
+    public var content: ElementContent {
+        return ElementContent(build: self.elementRepresentation(in:))
+    }
+
+    public func backingViewDescription(bounds: CGRect, subtreeExtent: CGRect?) -> ViewDescription? {
```

Comment:
Can we file this off as a GH issue to come back to later? It's definitely an interesting problem / thought.

Reply:
(A bit more context: With Workflow, we haven't been afraid to do large breaking refactors, which has been valuable to keep us moving forward – think we should take the same approach.)

Reply:
This would basically mean reversing `Element` and `ProxyElement`. Might be an improvement, but IMO hard to justify unless it's accompanied by some functional improvements in addition to the naming.